source/features/repo-wide-file-finder.tsx

```diff
@@ -0,0 +1,52 @@
+import './repo-wide-file-finder.css';
+import React from 'dom-chef';
+import features from '../libs/features';
+import select from 'select-dom';
+import {getRepoURL} from '../libs/utils';
+
+function init(): void {
+	const pageHeaderElement = select<HTMLUListElement>('.pagehead-actions');
+	const fileFinderButtonExists = select.exists('[data-hotkey="t"]');
+	const hiddenButton: HTMLElement = (
+		<li className="rgh-file-finder-hidden-btn">
+			<a
+				className="btn btn-sm empty-icon float-right BtnGroup-item"
+				data-hotkey="t"
+				data-pjax="true"
+				href={`/${getRepoURL()}/find/master`}
+			>
+				Hidden Find
+			</a>
+		</li>
+	);
+	if (!fileFinderButtonExists) {
+    pageHeaderElement?.appendChild(hiddenButton);
```

Comment:
This link doesn’t have to be public. We have two options:
- add it to `body` with the `hidden` attribute
- add it to `head` as a `link` rather than an anchor, but I don’t know if that’ll work

Reply:
Sorry, but I don't understand by what you mean by "This link doesn't have to be public". I'm creating a button and appending to only those pages where github doesn't already have the button. Could you kindly explain what advantage is expected by taking any of the approaches above?

Reply:
I meant: You’re appending it to a specific element and adding text inside it. If the link isn’t visible, you can just add it with 

```js
if (!select.exists('[data-hotkey="t"]') {
  document.body.append(<a hidden …/>);
}
```

The visual classes (`btn`), the text content and the `li` around it aren’t necessary. The user is not meant to see any of this.

Reply:
The code I showed is exactly all you need inside `init`. You can use `await getDefaultBranch()` directly inside the JSX tag, no need to create **any** variables. 

`deinit` isn’t necessary.